src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -131,10 +151,54 @@ std::shared_ptr<uint32_t> GetCounterHolder(const char * name)
 namespace chip {
 namespace Dnssd {
 
+std::string GetDomainFromHostName(const char * hostname)
+{
+    if (HostNameHasDomain(hostname, kLocalDot))
+    {
+        return std::string(kLocalDot);
+    }
+    else if (HostNameHasDomain(hostname, kOpenThreadDot))
+    {
+        return std::string(kOpenThreadDot);
+    }
+    return std::string();
```

Comment:
This looks wrong to me.  Why would we ignore all other domains?  Is this what Ted said to do?

Reply:
so Ted said the domain can be null or the open thread one but he was talking from his perspective. But we could have other valid domains so you are right. I wanted to keep the scope of the change small since previous code was always hardcoding to local. I will fix the getdomainfromhostname to do the right string operation and get the last .<domainname>. . I asked Ted if the open thread domain name should have a trailing dot. if not, then it will be a bit tricky.

Reply:
also will remove the hardcoding for matching with the local domain name and the open thread one.

Reply:
@bzbarsky-apple So when we get a browse with null domain we should browse both on local and the open thread. if we get an explicit domain we should browse on that domain. so my code needs changes. as for the resolve I think ted said the hostname should have a domain. I am double checking with him. if he confirms then for the resolve part, I will fix my GetDomainFromHostName api and get the domain from there. Does this sound good?

Reply:
modified this code to get the domain by operating on the string with an assumption the hostname looks like "<hostnameWithoutDomain>.<domain>." will add a comment in code for the same.